/*
 * Copyright 2022 Google LLC.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package yggdrasil_decision_forests.model.multitasker.proto;

import "yggdrasil_decision_forests/learner/abstract_learner.proto";

message MultitaskerTrainingConfig {
  // Learner for the sub-tasks.
  optional model.proto.TrainingConfig base_learner = 1;

  // Deployment for the sub-tasks.
  optional model.proto.DeploymentConfig base_learner_deployment = 2;

  // Sub tasks.
  repeated SubTask subtasks = 3;
}

message SubTask {
  // Definition of the sub tasks.
  optional model.proto.TrainingConfig train_config = 2;

  optional bool primary = 1 [default = true];
}

extend model.proto.TrainingConfig {
  optional MultitaskerTrainingConfig multitasker_config = 485868999;
}
